### PR TITLE
Add missing `uninstantiate(ancestor)`

### LINKED ISSUE #5281: Generics inheritance & initialization
This snippet fail to compile; note than without generic on `B(T)` it will works as expected

```crystal
module A
  @x : String

  def initialize(@x = "")
  end
end

class B(T)
  include A

  @y = {} of String => String
end

class C < B(String); end

C.new
```

Link to play the snippet:

https://play.crystal-lang.org/#/r/32r9

### PR DESCRIPTION
**Pull request by MakeNowJust**
Fixed #5281

### CODE REVIEW

**Comment by MakeNowJust** [comment]
Unfortunately [CI is failed](https://circleci.com/gh/crystal-lang/crystal/1787) on downloading LLVM. It is completely not related to this.

File: spec/compiler/semantic/instance_var_spec.cr [hunk 1]
```diff
@@ -4818,4 +4818,41 @@ describe "Semantic: instance var" do
...
+
+  it "can type ivar from class inherited by generic class (#5281)" do
+    assert_type(%(
+      class Foo
+        def initialize(@x = "foo")
+        end
+      end
+
+      class Bar(T) < Foo
+
```
Comment (Sija) [inline spec/compiler/semantic/instance_var_spec.cr]: minor: redundant line

Comment (MakeNowJust) [reply spec/compiler/semantic/instance_var_spec.cr]: Thank you!

**Comment by luislavena** [comment]
@mverzilli or @bcardiff, can we get your seal of approval on this?

Thank you.
❤️ ❤️ ❤️

**Pull request merged**